internal/controller/provisioner/objects.go

```diff
@@ -379,10 +406,18 @@ func (p *NginxProvisioner) buildNginxConfigMaps(
 		"Namespace":     p.cfg.GatewayPodConfig.Namespace,
 		"EnableMetrics": enableMetrics,
 		"MetricsPort":   metricsPort,
+		"AgentLabels":   telemetry.AgentLabelsToMap(p.cfg.AgentLabels),
 	}
 
 	if logging != nil && logging.AgentLevel != nil {
-		agentFields["LogLevel"] = *logging.AgentLevel
+		agentFields["LogLevel"] = logging.AgentLevel
+	}
+
+	if p.cfg.NginxOneConsoleTelemetryConfig.DataplaneKeySecretName != "" {
+		agentFields["DataplaneKeySecretName"] = p.cfg.NginxOneConsoleTelemetryConfig.DataplaneKeySecretName
```

Comment:
This field specifically could just be a boolean since it isn't rendered in the config. And maybe named something like "NginxOneReporting".